src/c_sources.ml

```diff
@@ -0,0 +1,115 @@
+open Stdune
+open Dune_file
+
+module Library = Dune_file.Library
+
+type t =
+  { libraries : C.Sources.t Lib_name.Map.t
+  }
+
+let for_lib t ~dir ~name =
+  match Lib_name.Map.find t.libraries name with
+  | Some m -> m
+  | None ->
+    Exn.code_error "C_sources.for_lib"
+      [ "name", Lib_name.to_sexp name
+      ; "dir", Path.to_sexp dir
+      ; "available", Sexp.Encoder.(list Lib_name.to_sexp)
+                       (Lib_name.Map.keys t.libraries)
+      ]
+
+let empty =
+  { libraries = Lib_name.Map.empty
+  }
+
+let c_name, cxx_name =
+  let make what ext ~loc s =
+    if match s with
+      | "" | "." | ".."  -> true
+      | _ -> false then
+      Errors.fail loc
+        "%S is not a valid %s name."
+        s what what ext
```

Comment:
The last two arguments are unused. I believe they are swallowed in the alpha returned by `Errors.fail`, damn...

Reply:
Haha, great catch. I recall there was a trick in base that made sure the arguments always ended with `()` to prevent these kinds of embarrassing bugs.